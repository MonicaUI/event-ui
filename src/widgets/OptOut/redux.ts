/* eslint-disable import/no-cycle */

import { injectPageSnapshot } from '../../redux/actions';
import querystring from 'querystring';

/**
 * Opt-out and opt-in Redux module:
 * Event invitees can click "Opt Out" link in their emails to opt them out all communications from the event account
 */
const GET_OPT_OUT_STATUS = 'event-guestside-site/optOut/GET_OPT_OUT_STATUS';
const OPT_OUT_CONTACT = 'event-guestside-site/optOut/OPT_OUT_CONTACT';

const createGetOptedOutStatusAction = contactInfo => {
  return {
    type: GET_OPT_OUT_STATUS,
    payload: contactInfo
  };
};

export const getOptOutStatus =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventGuestClient },
      accessToken,
      event: { id: eventId }
    } = getState();
    let {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    if (contactId) {
      inviteeId = null;
    }
    const optedOutContact = await eventGuestClient.getOptOutStatus(accessToken, eventId, inviteeId, guestId, contactId);
    dispatch(createGetOptedOutStatusAction(optedOutContact));
  };

export const createOptedOutAction = (contactInfo: $TSFixMe): $TSFixMe => {
  return {
    type: OPT_OUT_CONTACT,
    payload: contactInfo
  };
};

export const optOutAction =
  (optOutStatus: $TSFixMe, fromConfirmPage: $TSFixMe): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventGuestClient },
      accessToken,
      event: { id: eventId }
    } = getState();
    let {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    if (contactId) {
      inviteeId = null;
    }
    // eslint-disable-next-line import/no-cycle
    const page = (await import(/* webpackChunkName optOut */ './content')).optInOutPageNew;
    if ((fromConfirmPage && optOutStatus) || !fromConfirmPage) {
      const optedOutContact = await eventGuestClient.optOut(
        accessToken,
        eventId,
        inviteeId,
        guestId,
        contactId,
        optOutStatus
      );
      dispatch(createOptedOutAction(optedOutContact));
    }
    dispatch(injectPageSnapshot(page));
  };

export const prepareForOptOutPageLoad =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    let page;
    if (!inviteeId && !guestId && !contactId) {
      page = (await import(/* webpackChunkName optOut */ './invalidInviteeContent')).invalidInviteeOptOut;
    } else {
      await dispatch(getOptOutStatus());
      if (getState() && getState().optOut && getState().optOut.optOutStatus) {
        // eslint-disable-next-line import/no-cycle
        page = (await import(/* webpackChunkName optOut */ './content')).optInOutPage;
      } else {
        // eslint-disable-next-line import/no-cycle
        page = (await import(/* webpackChunkName optOut */ './content')).confirmationPage;
      }
    }
    dispatch(injectPageSnapshot(page));
  };

export const optOutReducer = (state = {}, action: $TSFixMe): $TSFixMe => {
  switch (action.type) {
    case GET_OPT_OUT_STATUS:
    case OPT_OUT_CONTACT:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};
