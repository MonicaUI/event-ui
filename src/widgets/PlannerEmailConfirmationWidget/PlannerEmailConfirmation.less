.checkboxPadding{
  padding: 15px;
}

.checkboxBorderPadding {
  width: 100%;
  margin-top: 5px;
  margin-bottom: 10px;
  padding: 12px 16px;
}

.checkboxOption {
  position: relative;
  left: 0;
  top: 0.25rem;
  -webkit-transform: translateY(-40%);
  transform: translateY(-40%);
}
