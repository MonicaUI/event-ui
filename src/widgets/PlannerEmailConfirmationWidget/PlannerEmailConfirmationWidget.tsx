import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BaseWidget from 'nucleus-widgets/lib/BaseWidget';
import { injectTestId } from '@cvent/nucleus-test-automation';
import Checkbox from 'nucleus-form/src/components/inputs/Checkbox';
import PlannerEmailConfirmationStyles from './PlannerEmailConfirmation.less';
import { merge } from 'lodash';
import { resolve } from '@cvent/nucleus-dynamic-css';
import { setSendEmailFlag } from '../../redux/registrationForm/regCart/actions';

const checkboxStyle = globalTheme => {
  return {
    ...globalTheme,
    checkbody1: {
      styleMapping: 'text2',
      customSettings: {
        spacing: {
          padding: {
            paddingTop: 0,
            paddingRight: 0,
            paddingBottom: 0,
            paddingLeft: 10
          }
        }
      }
    },
    checkbackground: {
      styleMapping: 'custom',
      customSettings: {
        background: {
          color: 'primary'
        }
      }
    },
    checkboxBorder: {
      styleMapping: 'custom',
      border: {
        borderStyle: 'solid',
        color: 'primary',
        thickness: 1
      },
      background: {
        color: 'secondary',
        borderRadius: 3
      }
    }
  };
};

/**
 * A widget to display a list of Attendees.
 */
class PlannerEmailConfirmationWidget extends BaseWidget<$TSFixMe, $TSFixMe> {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    style: PropTypes.any.isRequired,
    classes: PropTypes.any,
    sendEmail: (PropTypes as $TSFixMe).boolean,
    setSendEmailFlag: PropTypes.func
  };

  getElementInlineStyle: $TSFixMe;
  props: $TSFixMe;

  onChange = (fieldName, value, newOptionList) => {
    if (fieldName === 'sendEmail') {
      this.props.setSendEmailFlag(newOptionList[0].checked);
    }
  };

  render() {
    const { sendEmail, translate } = this.props;
    const checkboxOptions = [
      {
        name: translate('EventGuestSide_RegistrationNavigation_PlannerEmailConfirmation__resx'),
        checked: sendEmail,
        className: 'checkboxOption'
      }
    ];
    const styleObj = {
      classes: { ...PlannerEmailConfirmationStyles },
      checkbox: {
        listStyle: 'none'
      },
      horizontal: {
        checkboxLabel: { ...this.getElementInlineStyle('checkbody1'), ...this.getElementInlineStyle('checkbackground') }
      },
      checkboxBorder: this.getElementInlineStyle('checkboxBorder')
    };
    return (
      <div {...injectTestId('email-confirmation')} {...resolve(styleObj, 'checkboxPadding')}>
        <div {...resolve(styleObj, 'checkboxBorderPadding')} style={{ ...styleObj.checkboxBorder }}>
          <Checkbox
            fieldName="sendEmail"
            options={checkboxOptions}
            style={styleObj}
            classes={PlannerEmailConfirmationStyles}
            onChange={this.onChange}
          />
        </div>
      </div>
    );
  }
}

export default connect(
  (state: $TSFixMe) => {
    const {
      website: {
        theme: { global }
      },
      customFonts
    } = state;
    const style = merge({}, checkboxStyle(global), { customFonts });
    return {
      sendEmail: state.registrationForm.regCart.sendEmail,
      style
    };
  },
  { setSendEmailFlag }
)(PlannerEmailConfirmationWidget);
