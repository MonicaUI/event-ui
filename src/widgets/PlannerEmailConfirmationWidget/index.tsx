import React from 'react';

export function withPlannerEmailConfirmation(navigatorWidget: $TSFixMe, WrapperWidget: $TSFixMe) {
  return (props: $TSFixMe): $TSFixMe => <WrapperWidget {...props} NavigatorWidget={navigatorWidget} />;
}
