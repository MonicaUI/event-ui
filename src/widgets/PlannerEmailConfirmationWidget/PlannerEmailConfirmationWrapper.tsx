import React, { Suspense } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BaseWidget from 'nucleus-widgets/lib/BaseWidget';
import { REGISTRATION } from '../../redux/website/registrationProcesses';
import { getCurrentPageId } from '../../redux/pathInfo';

/**
 * A wrapper widget that wraps the Planner Email Confirmation and the Navigator widgets
 */
class PlannerEmailConfirmationWrapper extends BaseWidget<$TSFixMe, $TSFixMe> {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    style: PropTypes.any.isRequired,
    classes: PropTypes.any,
    isPlanner: (PropTypes as $TSFixMe).boolean,
    isLastPage: (PropTypes as $TSFixMe).boolean,
    NavigatorWidget: PropTypes.node
  };

  props: $TSFixMe;

  render() {
    const { isPlanner, isLastPage, NavigatorWidget } = this.props;
    const PlannerEmailConfirmationWidget = React.lazy(() => import('./PlannerEmailConfirmationWidget'));
    return (
      <div>
        {isLastPage && isPlanner && (
          <div>
            <Suspense fallback={<div></div>}>
              <PlannerEmailConfirmationWidget {...this.props} />
            </Suspense>
          </div>
        )}
        <NavigatorWidget {...this.props} />
      </div>
    );
  }
}

export default connect((state: $TSFixMe, props: $TSFixMe) => {
  const pageIds = REGISTRATION.forPathContainingWidget(props.id).pageIds(state);
  const currentPageId = getCurrentPageId(state);
  return {
    isPlanner: state.defaultUserSession.isPlanner,
    isLastPage:
      currentPageId.includes('registrationCancellationPage') ||
      currentPageId.includes('registrationDeclinePage') ||
      pageIds.indexOf(currentPageId) === pageIds.length - 1
  };
})(PlannerEmailConfirmationWrapper);
