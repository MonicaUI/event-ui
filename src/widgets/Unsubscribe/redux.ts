import { injectPageSnapshot } from '../../redux/actions';
import querystring from 'querystring';

/**
 * Unsubscribe and re-subscribe Redux module:
 * Event invitees can click "Unsubscribe" link in their emails to opt them out from the event emails communication
 */
const GET_UNSUBSCRIBE_STATUS = 'event-guestside-site/unsubscribe/GET_UNSUBSCRIBE_STATUS';
const SET_UNSUBSCRIBE_STATUS = 'event-guestside-site/unsubscribe/SET_UNSUBSCRIBE_STATUS';

const createGetUnsubscribeStatusAction = (unsubscribedContact, optOutStatus) => {
  return {
    type: GET_UNSUBSCRIBE_STATUS,
    payload: {
      ...optOutStatus,
      unsubscribeStatus: !unsubscribedContact.subscribed
    }
  };
};

export const getUnsubscribeStatus =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventEmailClient, eventGuestClient },
      accessToken,
      event: { id: eventId },
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const encodedGuestId = queryParams.g;
    const unsubscribedContact = await eventEmailClient.getSubscriptionStatus(eventId, inviteeId, encodedGuestId);
    // optOutStatus object has invitee/guest and planner infomation for display
    const optOutStatus = await eventGuestClient.getOptOutStatus(accessToken, eventId, inviteeId, encodedGuestId);
    dispatch(createGetUnsubscribeStatusAction(unsubscribedContact, optOutStatus));
  };

const createUnsubscribeAction = unsubscribeStatus => {
  return {
    type: SET_UNSUBSCRIBE_STATUS,
    payload: { unsubscribeStatus }
  };
};

export const unsubscribeAction =
  (unsubscribeStatus: $TSFixMe): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventEmailClient },
      event: { id: eventId },
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const encodedGuestId = queryParams.g;
    const subscribed = !unsubscribeStatus;
    await eventEmailClient.setSubscriptionStatus(eventId, inviteeId, encodedGuestId, subscribed);
    dispatch(createUnsubscribeAction(unsubscribeStatus));
  };

export const prepareForUnsubscribePageLoad =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const encodedGuestId = queryParams.g;
    let unsubscribePage;
    if (!inviteeId && !encodedGuestId) {
      unsubscribePage = (await import(/* webpackChunkName: "unsubscribe" */ './invalidInviteeContent'))
        .invalidInviteeUnsubscribePage;
    } else {
      await dispatch(getUnsubscribeStatus());
      // eslint-disable-next-line import/no-cycle
      unsubscribePage = (await import(/* webpackChunkName: "unsubscribe" */ './content')).unsubscribePage;
    }
    dispatch(injectPageSnapshot(unsubscribePage));
  };

export const unsubscribeReducer = (state = {}, action: $TSFixMe): $TSFixMe => {
  switch (action.type) {
    case GET_UNSUBSCRIBE_STATUS:
    case SET_UNSUBSCRIBE_STATUS:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};
