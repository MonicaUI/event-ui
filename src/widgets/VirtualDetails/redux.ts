import querystring from 'querystring';
import { injectPageSnapshot } from '../../redux/actions';
import { populateAllProducts } from '../../redux/visibleProducts';
import { isBeforeStartTime, currentTimeIsInRangeOf } from 'event-widgets/redux/modules/timezones';
import { CANCELLED } from 'event-widgets/clients/EventStatus';
import { sessionStatus } from 'event-widgets/redux/selectors/event';
import Logger from '@cvent/nucleus-logging';

const SET_VIRTUAL_DETAILS = 'event-guestside-site/virtualDetails/SET_VIRTUAL_DETAILS';
const DISPLAY_SESSION_NOT_AVAILABLE_MESSAGE =
  'event-guestside-site/virtualDetails/DISPLAY_SESSION_NOT_AVAILABLE_MESSAGE';
export const EVENT = 'event';
export const SESSION = 'session';
export const VIRTUAL_DETAILS_PAGE = 'virtualDetailsPage';
export const SESSION_NOT_AVAILABLE_PAGE = 'sessionNotAvailableMessagePage';
const LOG = new Logger('widgets/virtualDetails/redux');

const setVirtualDetails = ({ code, url, type, isPassed, pageToDisplay }) => {
  return {
    type: SET_VIRTUAL_DETAILS,
    payload: { code, url, type, isPassed, pageToDisplay }
  };
};

const displaySessionNotAvailableMessage = () => {
  return {
    type: DISPLAY_SESSION_NOT_AVAILABLE_MESSAGE,
    payload: { pageToDisplay: SESSION_NOT_AVAILABLE_PAGE }
  };
};

export const joinWebcastSession =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      virtualDetails: { url }
    } = getState();
    if (url) {
      window.open(url, '_blank');
    }
  };

const getURLAndCode = (attendeeWebcastData, genericWebcastData) => {
  let url;
  let code;

  if (attendeeWebcastData?.join?.href) {
    url = attendeeWebcastData?.join?.href;
    code = attendeeWebcastData?.join?.code;
  } else if (genericWebcastData?.joiningURL) {
    url = genericWebcastData?.joiningURL;
    code = genericWebcastData?.joiningCode;
  }

  return { url, code };
};

const fetchSessionWebcastData = (attendeeId, sessionId) => async (dispatch, getState) => {
  let {
    // eslint-disable-next-line prefer-const
    selectedTimeZone,
    // eslint-disable-next-line prefer-const
    text: { eventTimezone },
    // eslint-disable-next-line prefer-const
    clients: { universalWebcastClient },
    visibleProducts: { Widget: { sessionProducts = {} } = {} }
  } = getState();
  let virtualData = {};
  let attendeeLinksContainer;
  let attendeeWebcastData;

  await dispatch(populateAllProducts('Widget'));
  sessionProducts = getState().visibleProducts?.Widget?.sessionProducts;
  const selectedSessionDetails = sessionProducts[sessionId];

  // If session was deleted in test mode
  if (!selectedSessionDetails) {
    dispatch(displaySessionNotAvailableMessage());
    return;
  }

  if (selectedSessionDetails.status === sessionStatus.cancelled) {
    return;
  }

  const isSessionPassed = !(
    currentTimeIsInRangeOf(
      selectedSessionDetails.startTime,
      selectedSessionDetails.endTime,
      selectedTimeZone,
      eventTimezone
    ) || isBeforeStartTime(selectedSessionDetails.startTime, selectedTimeZone, eventTimezone)
  );

  if (isSessionPassed) {
    if (selectedSessionDetails?.webcast?.recordingURL) {
      virtualData = {
        url: selectedSessionDetails?.webcast?.recordingURL,
        code: selectedSessionDetails?.webcast.recordingCode
      };
    } else {
      return;
    }
  } else {
    try {
      attendeeLinksContainer = await universalWebcastClient.getAttendeeLinkData(attendeeId);
    } catch (error) {
      LOG.warn(`Failed to get attendee specific session webcast data for attendeeId ${attendeeId}`);
      return;
    }
    attendeeWebcastData = attendeeLinksContainer?.data.find(
      attendeeLinkData => attendeeLinkData?.session?.id === sessionId
    );
    virtualData = getURLAndCode(attendeeWebcastData, selectedSessionDetails.webcast);
  }

  if (!(virtualData as $TSFixMe)?.url) {
    return;
  }

  dispatch(
    // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ type: string; isPassed: boolea... Remove this comment to see the full error message
    setVirtualDetails({ ...virtualData, type: SESSION, isPassed: isSessionPassed, pageToDisplay: VIRTUAL_DETAILS_PAGE })
  );
};

const fetchEventWebcastData = attendeeId => async (dispatch, getState) => {
  const {
    event: { status: eventStatus, webcast: eventWebcast, startDate, endDate },
    selectedTimeZone,
    text: { eventTimezone },
    clients: { universalWebcastClient }
  } = getState();
  let virtualData = {};
  let attendeeLinksContainer;
  let attendeeWebcastData;

  if (eventStatus === CANCELLED) {
    return;
  }

  const isEventPassed = !(
    currentTimeIsInRangeOf(startDate, endDate, selectedTimeZone, eventTimezone) ||
    isBeforeStartTime(startDate, selectedTimeZone, eventTimezone)
  );

  if (isEventPassed) {
    if (eventWebcast?.recordingURL) {
      virtualData = {
        url: eventWebcast?.recordingURL,
        code: eventWebcast?.recordingCode
      };
    } else {
      return;
    }
  } else {
    try {
      attendeeLinksContainer = attendeeId ? await universalWebcastClient.getAttendeeLinkData(attendeeId) : undefined;
    } catch (error) {
      LOG.warn(`Failed to get attendee specific event webcast data for attendeeId ${attendeeId}`);
      return;
    }
    attendeeWebcastData = attendeeLinksContainer?.data.find(
      attendeeLinkData => Object.keys(attendeeLinkData?.session).length === 0
    );
    virtualData = getURLAndCode(attendeeWebcastData, eventWebcast);
  }

  if (!(virtualData as $TSFixMe)?.url) {
    return;
  }

  dispatch(
    // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ type: string; isPassed: boolea... Remove this comment to see the full error message
    setVirtualDetails({ ...virtualData, type: EVENT, isPassed: isEventPassed, pageToDisplay: VIRTUAL_DETAILS_PAGE })
  );
};

export const prepareForVirtualDetailsPageLoad =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const sessionId = queryParams.sessionId;
    const attendeeId = guestId || inviteeId;
    let virtualDetailsPage;

    if (sessionId) {
      if (attendeeId) {
        await dispatch(fetchSessionWebcastData(attendeeId, sessionId));
      }
    } else {
      await dispatch(fetchEventWebcastData(attendeeId));
    }

    if (getState()?.virtualDetails?.pageToDisplay === VIRTUAL_DETAILS_PAGE) {
      // eslint-disable-next-line import/no-cycle
      virtualDetailsPage = (await import(/* webpackChunkName: "virtualDetails" */ './content')).virtualDetailsPage;
    } else if (getState()?.virtualDetails?.pageToDisplay === SESSION_NOT_AVAILABLE_PAGE) {
      // eslint-disable-next-line import/no-cycle
      virtualDetailsPage = (await import(/* webpackChunkName: "virtualDetails" */ './content'))
        .sessionNotAvailableMessagePage;
    } else {
      // eslint-disable-next-line import/no-cycle
      virtualDetailsPage = (await import(/* webpackChunkName: "virtualDetails" */ './content')).genericMessagePage;
    }

    dispatch(injectPageSnapshot(virtualDetailsPage));
  };

export const virtualDetailsReducer = (state = {}, action: $TSFixMe): $TSFixMe => {
  switch (action.type) {
    case SET_VIRTUAL_DETAILS:
    case DISPLAY_SESSION_NOT_AVAILABLE_MESSAGE:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};
