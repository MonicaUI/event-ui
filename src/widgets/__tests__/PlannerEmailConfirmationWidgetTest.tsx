import React from 'react';
import renderer from 'react-test-renderer';
import PlannerEmailConfirmationWidget from '../PlannerEmailConfirmationWidget/PlannerEmailConfirmationWidget';
import { createStore } from 'redux';

jest.mock('../../redux/registrationForm/regCart/actions', () => {
  return {
    setSendEmailFlag: jest.fn(x => x)
  };
});
jest.mock('../../redux/pathInfo', () => {
  return {
    getCurrentPageId: () => {
      return '12345';
    }
  };
});

const initialState = {
  event: {},
  text: {
    translate: () => jest.fn(x => x)
  },
  guestText: {
    translate: () => jest.fn(x => x)
  },
  defaultUserSession: {
    isPlanner: true
  },
  website: {
    theme: {
      global: {
        elements: {}
      }
    }
  },
  customFonts: {},
  registrationForm: {
    regCart: {
      sendEmail: true
    }
  }
};

const defaultProps = {
  classes: {},
  style: {},
  translate: jest.fn(x => x),
  config: {
    displayText: {
      backward: 'backward',
      forward: 'forward',
      complete: 'complete',
      exit: 'exit'
    }
  }
};

describe('PlannerEmailConfirmationWidget', () => {
  test('should render', () => {
    const widget = renderer.create(
      <PlannerEmailConfirmationWidget
        {...defaultProps}
        // @ts-expect-error ts-migrate(2322) FIXME: Type '{ store: Store<unknown, Action<any>>; classe... Remove this comment to see the full error message
        store={createStore((state = initialState) => {
          return state;
        })}
      />
    );
    expect(widget).toMatchSnapshot();
  });
});
