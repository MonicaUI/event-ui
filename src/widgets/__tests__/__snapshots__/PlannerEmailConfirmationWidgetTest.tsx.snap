// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlannerEmailConfirmationWidget should render 1`] = `
<div
  className=""
  data-cvent-id="email-confirmation"
  style={Object {}}
>
  <div
    className=""
    style={
      Object {
        "backgroundColor": "secondary",
        "borderColor": "primary",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 1,
        "hover": "css-87ndg2",
      }
    }
  >
    <fieldset
      className=""
      style={Object {}}
    >
      <legend
        className=""
        htmlFor="sendEmail"
        style={Object {}}
      >
        <span
          className=""
          style={Object {}}
        >
          <span
            data-cvent-id="label"
          />
        </span>
      </legend>
      <div
        className=""
        style={Object {}}
      >
        <ul
          className=""
          data-cvent-id="checkbox"
          id="sendEmail"
          style={
            Object {
              "checkboxLabel": Object {},
              "listStyle": "none",
            }
          }
        >
          <li
            data-cvent-id="option-sendEmail_0"
          >
            <input
              checked={true}
              className=""
              id="sendEmail_0"
              name="sendEmail"
              onChange={[Function]}
              required={false}
              style={Object {}}
              type="checkbox"
              value="0"
            />
            <label
              htmlFor="sendEmail_0"
              style={Object {}}
            >
              EventGuestSide_RegistrationNavigation_PlannerEmailConfirmation__resx
            </label>
          </li>
        </ul>
      </div>
    </fieldset>
  </div>
</div>
`;
